app/src/main/java/edu/temple/templetag/fragments/MapFragment.java

```diff
@@ -37,41 +37,56 @@
 import edu.temple.templetag.HomeActivity;
 import edu.temple.templetag.R;
 import edu.temple.templetag.Tag;
+import edu.temple.templetag.tools.DistanceCalculator;
 
 public class MapFragment extends Fragment implements OnMapReadyCallback {
     private static final String ARG_PARAM1 = "tags";
+    private static final String ARG_PARAM3 = "tag";
     private static final String ARG_PARAM2 = "currentLocation";
-    private static final double MAX_RADIUS = 10;
     private Location currentLocation;
+    private DistanceCalculator distanceCalculator = new DistanceCalculator();
 
     // TODO: Rename and change types of parameters
     private ArrayList<Tag> tags;
+    private Tag tag;
 
     //Initiate Map Stuff
     MapView mapView;
-    GoogleMap googleMap;
+    public GoogleMap googleMap;
 
     public MapFragment() {
         // Required empty public constructor
     }
 
 
-    public static MapFragment newInstance() {
+    public static MapFragment newInstance(ArrayList<Tag> tags, Location currentLocation) {
         MapFragment fragment = new MapFragment();
-//        Bundle args = new Bundle();
-//        args.putParcelableArrayList(ARG_PARAM1, tags);
-//        args.putParcelable(ARG_PARAM2, currentUserLocation);
-//        fragment.setArguments(args);
+        Bundle args = new Bundle();
+        args.putParcelableArrayList(ARG_PARAM1, tags);
+        args.putParcelable(ARG_PARAM2, currentLocation);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    public static MapFragment newInstance(Tag tag, Location currentLocation) {
```

Comment:
Are we passing one tag into the MapFragment or should it be an ArrayList<Tag>?

Reply:
From HomeActivity we will pass an ArrayList but in tagDetailActivity, we only need to pass one tag